HPPAMapping.c:422:43: error: comparison between ‘hppa_reg’ and ‘enum hppa_op_type’ [-Werror=enum-compare]

@R33v0LT I am building it as part of Rizin and due to the stricter compiler flags, it errors there, please take a look.
```
../subprojects/capstone-next/arch/HPPA/HPPAMapping.c: In function ‘HPPA_reg_access’:
../subprojects/capstone-next/arch/HPPA/HPPAMapping.c:422:43: error: comparison between ‘hppa_reg’ and ‘enum hppa_op_type’ [-Werror=enum-compare]
  422 |                         if (op->mem.space != HPPA_OP_INVALID)
      |                                           ^~
cc1: all warnings being treated as errors
[15/2016] Compiling C object subprojects/capstone-next/libcapstone.a.p/arch_HPPA_HPPAInstPrinter.c.o
FAILED: subprojects/capstone-next/libcapstone.a.p/arch_HPPA_HPPAInstPrinter.c.o 
gcc -Isubprojects/capstone-next/libcapstone.a.p -I../subprojects/capstone-next/include -I../subprojects/capstone-next/include/capstone -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O3 --std=gnu99 -Werror=sizeof-pointer-memaccess -fvisibility=hidden -DRZ_ASSERT_STDOUT=1 -Wno-cpp -fPIC -DCAPSTONE_X86_ATT_DISABLE_NO -DCAPSTONE_X86_REDUCE_NO -DCAPSTONE_USE_SYS_DYN_MEM -DCAPSTONE_DIET_NO -DCAPSTONE_HAS_ALPHA -DCAPSTONE_HAS_ARM -DCAPSTONE_HAS_AARCH64 -DCAPSTONE_HAS_HPPA -DCAPSTONE_HAS_M68K -DCAPSTONE_HAS_M680X -DCAPSTONE_HAS_MIPS -DCAPSTONE_HAS_POWERPC -DCAPSTONE_HAS_SPARC -DCAPSTONE_HAS_SYSZ -DCAPSTONE_HAS_X86 -DCAPSTONE_HAS_XCORE -DCAPSTONE_HAS_TMS320C64X -DCAPSTONE_HAS_TRICORE -Wno-maybe-uninitialized -MD -MQ subprojects/capstone-next/libcapstone.a.p/arch_HPPA_HPPAInstPrinter.c.o -MF subprojects/capstone-next/libcapstone.a.p/arch_HPPA_HPPAInstPrinter.c.o.d -o subprojects/capstone-next/libcapstone.a.p/arch_HPPA_HPPAInstPrinter.c.o -c ../subprojects/capstone-next/arch/HPPA/HPPAInstPrinter.c
../subprojects/capstone-next/arch/HPPA/HPPAInstPrinter.c: In function ‘print_operand’:
../subprojects/capstone-next/arch/HPPA/HPPAInstPrinter.c:612:35: error: comparison between ‘hppa_reg’ and ‘enum hppa_op_type’ [-Werror=enum-compare]
  612 |                 if (op->mem.space != HPPA_OP_INVALID &&
      |                                   ^~
```
For the reference: https://github.com/rizinorg/rizin/pull/4394

--- a/arch/HPPA/HPPAInstPrinter.c
+++ b/arch/HPPA/HPPAInstPrinter.c
@@ -609,7 +609,7 @@ static void print_operand(MCInst *MI, SStream *O, const cs_hppa_op *op)
 		break;
 	case HPPA_OP_MEM:
 		SStream_concat(O, "(");
-		if (op->mem.space != HPPA_OP_INVALID &&
+		if (op->mem.space != HPPA_REG_INVALID &&
 		    op->mem.space != HPPA_REG_SR0) {
 			SStream_concat(O, HPPA_reg_name((csh)MI->csh,
 							op->mem.space));

--- a/arch/HPPA/HPPAMapping.c
+++ b/arch/HPPA/HPPAMapping.c
@@ -419,7 +419,7 @@ void HPPA_reg_access(const cs_insn *insn, cs_regs regs_read,
 			}
 			break;
 		case HPPA_OP_MEM:
-			if (op->mem.space != HPPA_OP_INVALID)
+			if (op->mem.space != HPPA_REG_INVALID)
 				regs_read[read_count++] = op->mem.space;
 			if (op->mem.base_access & CS_AC_READ) {
 				regs_read[read_count++] = op->mem.base;

--- a/bindings/python/test_hppa.py
+++ b/bindings/python/test_hppa.py
@@ -42,7 +42,7 @@ def print_insn_detail(insn):
                 print("\t\toperands[%u].type: DISP = 0x%s" % (c, to_x(i.imm)))
             if i.type == HPPA_OP_MEM:
                 print("\t\toperands[%u].type: MEM" % c)
-                if i.mem.space != HPPA_OP_INVALID:
+                if i.mem.space != HPPA_REG_INVALID:
                     print("\t\t\toperands[%u].mem.space: REG = %s" % (c, insn.reg_name(i.mem.space)))
                 print("\t\t\toperands[%u].mem.base: REG = %s" % (c, insn.reg_name(i.mem.base)))
             if i.type == HPPA_OP_TARGET:

--- a/cstool/cstool_hppa.c
+++ b/cstool/cstool_hppa.c
@@ -51,7 +51,7 @@ void print_insn_detail_hppa(csh handle, cs_insn *ins)
 			break;
 		case HPPA_OP_MEM:
 			printf("\t\toperands[%u].type:  MEM\n", i);
-			if (op->mem.space != HPPA_OP_INVALID) {
+			if (op->mem.space != HPPA_REG_INVALID) {
 				printf("\t\t\toperands[%u].mem.space: REG = %s\n",
 				       i, cs_reg_name(handle, op->mem.space));
 			}

--- a/suite/cstest/src/hppa_detail.c
+++ b/suite/cstest/src/hppa_detail.c
@@ -48,7 +48,7 @@ char *get_detail_hppa(csh *p_handle, cs_mode mode, cs_insn *ins)
 			break;
 		case HPPA_OP_MEM:
 			add_str(&result, "\t\toperands[%u].type:  MEM\n", i);
-			if (op->mem.space != HPPA_OP_INVALID) {
+			if (op->mem.space != HPPA_REG_INVALID) {
 				add_str(&result,
 					"\t\t\toperands[%u].mem.space: REG = %s\n",
 					i, cs_reg_name(handle, op->mem.space));

--- a/tests/test_hppa.c
+++ b/tests/test_hppa.c
@@ -64,7 +64,7 @@ static void print_insn_detail(cs_insn *ins)
 			break;
 		case HPPA_OP_MEM:
 			printf("\t\toperands[%u].type: MEM\n", i);
-			if (op->mem.space != HPPA_OP_INVALID) {
+			if (op->mem.space != HPPA_REG_INVALID) {
 				printf("\t\t\toperands[%u].mem.space: REG = %s\n",
 				       i, cs_reg_name(handle, op->mem.space));
 			}

